src/main.rs

```diff
@@ -38,46 +41,36 @@ fn main() {
     let ui = MainWindow::new().unwrap();
 
     // === Sign in, if user has a token ===
-    if !(*auth_store).borrow().is_empty() {
-        fetch_data(&ui, &auth_store.clone());
+    if !auth_store.is_empty() {
+        auth_store.retain_and_rewrite(|auth| {
```

Comment:
Result will never return anything other then an error, so `_ => {...` will never get triggred

Reply:
it does get triggered IF status code doesnt equal unathorized but I think i will just put if statements inside an error arm-binding